src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequestFactories.java

```diff
@@ -277,6 +278,54 @@ public NBTTagCompound serialize(@NotNull final IFactoryController controller, @N
         }
     }
 
+    @SuppressWarnings(Suppression.BIG_CLASS)
+    public static final class SmeltableOreRequestFactory implements IRequestFactory<SmeltableOre, StandardRequests.SmeltAbleOreRequest>
+    {
+
+        @Override
+        public StandardRequests.SmeltAbleOreRequest getNewInstance(
+                @NotNull final SmeltableOre input,
+                @NotNull final IRequester location,
+                @NotNull final IToken token,
+                @NotNull final RequestState initialState)
+        {
+            return new StandardRequests.SmeltAbleOreRequest(location, token, initialState, input);
+        }
+
+        @NotNull
+        @Override
+        public TypeToken<? extends StandardRequests.SmeltAbleOreRequest> getFactoryOutputType()
+        {
+            return TypeToken.of(StandardRequests.SmeltAbleOreRequest.class);
+        }
+
+        @NotNull
+        @Override
+        public TypeToken<? extends SmeltableOre> getFactoryInputType()
+        {
+            return TypeToken.of(SmeltableOre.class);
+        }
+
+        @NotNull
+        @Override
+        public NBTTagCompound serialize(@NotNull final IFactoryController controller, @NotNull final StandardRequests.SmeltAbleOreRequest request)
+        {
+            return serializeToNBT(controller, request, SmeltableOre::serialize);
+        }
+
+        @NotNull
+        @Override
+        public StandardRequests.SmeltAbleOreRequest deserialize(@NotNull final IFactoryController controller, @NotNull final NBTTagCompound nbt)
+        {
+            return deserializeFromNBT(controller, nbt, SmeltableOre::deserialize,
+                    (requested, token, requester, requestState) -> controller.getNewInstance(TypeToken.of(StandardRequests.SmeltAbleOreRequest.class),
+                            requested,
+                            token,
+                            requester,
+                            requestState));
+        }
+    }
+
     @SuppressWarnings(Suppression.BIG_CLASS)
     public static final class BurnableRequestFactory implements IRequestFactory<Burnable, StandardRequests.BurnableRequest>
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 42 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)